src/jit/lsraxarch.cpp

```diff
@@ -2530,81 +2588,74 @@ void LinearScan::BuildIndir(GenTreeIndir* indirTree)
 // Return Value:
 //    None.
 //
-void LinearScan::BuildMul(GenTree* tree)
+int LinearScan::BuildMul(GenTree* tree)
 {
-    TreeNodeInfo* info = currentNodeInfo;
-#if defined(_TARGET_X86_)
-    assert(tree->OperIs(GT_MUL, GT_MULHI, GT_MUL_LONG));
-#else
-    assert(tree->OperIs(GT_MUL, GT_MULHI));
-#endif
-    GenTree* op1   = tree->gtOp.gtOp1;
-    GenTree* op2   = tree->gtOp.gtOp2;
-    info->srcCount = appendBinaryLocationInfoToList(tree->AsOp());
-    assert(info->dstCount == 1);
+    assert(tree->OperIsMul());
+    GenTree* op1 = tree->gtOp.gtOp1;
+    GenTree* op2 = tree->gtOp.gtOp2;
 
-    // Case of float/double mul.
-    if (varTypeIsFloating(tree->TypeGet()))
-    {
-        return;
-    }
+    int       srcCount      = BuildRMWUses(tree->AsOp());
+    regMaskTP dstCandidates = RBM_NONE;
 
-    bool isUnsignedMultiply    = ((tree->gtFlags & GTF_UNSIGNED) != 0);
-    bool requiresOverflowCheck = tree->gtOverflowEx();
+    // Only non-floating point mul has special requirements
+    if (!varTypeIsFloating(tree->TypeGet()))
```

Comment:
Make it consistent with BuildDivMod that does `if (is float) { build simple; return; }`? Or move the float check inside the switch case and rename BuildMul and BuildDivMod to BuildIntegerMul and BuildIntegerDivMod?

Note that FP add/sub/mul/div use the same instruction format but they're handled more or less differently throughout lower/ra/codegen. I have a PR to clean this up but I was waiting for your lsra refactoring to finish it.